### PR TITLE
Added namespace tools module

### PR DESCRIPTION
**Pull request by pllim**
Added namespace tools module.

This is a solution for:

* Finding data files in shared namespace (see #166 )
* Finding custom plugins in shared namespace (see #92) -- Can only find them, cannot auto-load them

### CODE REVIEW

File: ginga/util/nstools.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
+#
+# Copyright (c) Eric R. Jeschke.  All rights reserved.
+# This is open-source software licensed under a BSD license.
+# Please see the file LICENSE.txt for details.
+#
+"""Tools for shared Ginga namespace.
+
+These can be used by other developer writing plugins
+under the same Ginga namespace but in a separate package.
```
Comment (pllim) [inline ginga/util/nstools.py]: Perhaps more suitable to be part of `ginga/misc/PluginManager.py`?

File: ginga/util/nstools.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    # No custom plugins
+    if len(paths) < 2:
+        return plugins
+
+    # Identify core distribution path
+    coremod = __import__('ginga.Control')  # Only core has this?
+    corepath = os.path.dirname(coremod.__file__)
+
+    # Find custom plugins in shared namespace.
+    # TODO: How to separate global and local plugins?
```
Comment (pllim) [inline ginga/util/nstools.py]: If we can figure out how to identify and separate global and local plugins, perhaps this is better off as a method within `PluginManager` class?

File: ginga/util/nstools.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            if pfx in plugins:
+                plugins[pfx].union(pnames)
+            else:
+                plugins[pfx] = pnames
+
+    # Search HOME (*py and *pyc give duplicates)
+    if include_home:
+        home = os.path.join(os.path.expanduser('~'), '.ginga', 'plugins')
+        if os.path.isdir(home):
+            pnames = set(os.path.splitext(fn)[0] for fn in os.listdir(home))
```
Comment (pllim) [inline ginga/util/nstools.py]: If this becomes a method, the repeated incantation of finding `pnames` and inserting it into the dictionary can be a hidden method and is called here?

**Comment by pllim** [comment]
@eteq, the "auto discovery" function is in this PR. FYI.

File: ginga/util/nstools.py (hunk 2 above)
Comment (ejeschke) [inline ginga/util/nstools.py]: Well, they do have different constructor signatures.  We probably can use python's `inspect` module to determine whether it is a local or global plugin.

**Comment by pllim** [comment]
It appears that STScI decided to not use shared namespace after all as per spacetelescope/stginga#18 . Do you still want to pursue this in case others are interested, or should we close this as "no go"?

**Comment by ejeschke** [comment]
If STScI doesn't need this now then I think we can just close it.  I suspect it will resurface again as an issue, but maybe then I can learn from your experience.

**Comment by pllim** [comment]
Do you want me to revert #169 as well?

**Comment by ejeschke** [comment]
No, I think we'll leave that one for now.

**Comment by pllim** [comment]
Okay.

**Pull request closed without merging**